### PR TITLE
Twig template for projects' tasks

### PR DESCRIPTION
**Pull request by AdrienClairembault**
Viewing projects' tasks from the "Your planning" widget of the  "Personal view" tab of the home page is broken:

image

It seems to be outdated table based html that isn't really responsive and can't deal with the smaller available width.

I've migrated it to twig to fix the issue.

After fix:

image

And here the before/after from the project task main tab:

image

image

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !26056

### CODE REVIEW

File: src/ProjectTask.php [hunk 1]
```diff
@@ -583,315 +584,45 @@ public static function getAllTicketsForProject($ID)
...
-        $rand_milestone          = mt_rand();
-        $rand_plan_start_date    = mt_rand();
-        $rand_plan_end_date      = mt_rand();
-        $rand_real_start_date    = mt_rand();
-        $rand_real_end_date      = mt_rand();
-        $rand_effective_duration = mt_rand();
-        $rand_planned_duration   = mt_rand();
-
         if ($ID > 0) {
+            $duration = ProjectTask_Ticket::getTicketsTotalActionTime($this->getID());
```
Comment (cedric-anne) [inline src/ProjectTask.php:588]: Should be placed after next line, to save a useless DB query if check does not pass.

File: src/ProjectTask.php [hunk 2]
```diff
@@ -583,315 +584,45 @@ public static function getAllTicketsForProject($ID)
...
             $this->check($ID, READ);
             $projects_id     = $this->fields['projects_id'];
             $projecttasks_id = $this->fields['projecttasks_id'];
+            $recursive       = null; // Wont be used in the case, value is irrelevant
+            $ticket_duration = Html::timestampToString($duration, false);
+            $total_duration  = Html::timestampToString(
+                $duration + $this->fields["effective_duration"],
+                false
+            );
```
Comment (cedric-anne) [inline src/ProjectTask.php:597]: Should be formatted on template side using `|formatted_duration(false)` filter. Filter will have to be updated to get an optional `bool $display_seconds = true` parameter.

File: src/ProjectTask.php [hunk 3]
```diff
@@ -583,315 +584,45 @@ public static function getAllTicketsForProject($ID)
...
+            'item'                     => $this,
+            'params'                   => $options,
+            'parent'                   => Project::getById($projects_id),
+            'projects_id'              => $projects_id,
+            'projecttasks_id'          => $projecttasks_id,
+            'recursive'                => $recursive,
+            'duration_dropdown_to_add' => $duration_dropdown_to_add,
+            'ticket_duration'          => $ticket_duration,
+            'total_duration'           => $total_duration,
+            'safe_content'             => RichText::getSafeHtml($this->fields["content"], true),
```
Comment (cedric-anne) [inline src/ProjectTask.php:624]: This should not be needed. Indeed, `|safe_html|escape` filters are already applied by textarea field macro.

```suggestion
            'content'             => $this->fields["content"],
```

File: templates/pages/tools/project_task.html.twig [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+{% import 'components/form/fields_macros.html.twig' as fields %}
+{% set params  = params ?? [] %}
+
+{% block form_fields %}
+
+    {{ fields.dropdownField(
+        'ProjectTaskTemplate',
+        'projecttasktemplates_id',
+        item.fields.projecttasktemplates_id,
+        _n('Project task template', 'Project task templates', 1),
```
Comment (cedric-anne) [inline templates/pages/tools/project_task.html.twig:44]:
```suggestion
        'ProjectTaskTemplate'|itemtype_name,
```

File: templates/pages/tools/project_task.html.twig [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+            $.ajax({
+                url: CFG_GLPI.root_doc + "/ajax/projecttask.php",
+                type: "POST",
+                data: {
+                    projecttasktemplates_id: value
+                }
+            }).done(function(data) {
+                // Set simple inputs
+                $('input[name=name]').val(data.name);
+                $('textarea[name=comment]').val(data.comments);
```
Comment (cedric-anne) [inline templates/pages/tools/project_task.html.twig:62]: Selectors are not precise enough, `name` and `comment` are common field names. You should either find a way to trigger the form.
Maybe something like:
```js
function projecttasktemplate_update(value) {
    const form = $(this).closest('form');
```
and then:
```js
form.find('input[name=name]').val(data.name);
```

File: templates/pages/tools/project_task.html.twig [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+    {{ fields.checkboxField(
+        'auto_percent_done',
+        item.fields.auto_percent_done,
+        __('Automatically calculate'),
+        {
+            'add_field_html': tooltip,
+        }
+    ) }}
+    <script>
+        $("input[name=auto_percent_done]").on('change', function() {
```
Comment (cedric-anne) [inline templates/pages/tools/project_task.html.twig:164]: Selectors based on random ID would be safer here.

File: templates/pages/tools/project_task.html.twig [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+        });
+    </script>
+
+    {{ fields.dropdownYesNo(
+        'is_milestone',
+        item.fields.is_milestone,
+        __('Milestone'),
+    ) }}
+    <script>
+        $('select[name=is_milestone]').on('change', function() {
```
Comment (cedric-anne) [inline templates/pages/tools/project_task.html.twig:175]: Same here.

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**